nitransforms/io/lta.py

```diff
@@ -24,26 +24,23 @@ class VolumeGeometry(StringBasedStruct):
     template_dtype = np.dtype(
         [
             ("valid", "i4"),  # Valid values: 0, 1
-            ("volume", "i4", (3, 1)),  # width, height, depth
-            ("voxelsize", "f4", (3, 1)),  # xsize, ysize, zsize
+            ("volume", "i4", (3, )),  # width, height, depth
+            ("voxelsize", "f4", (3, )),  # xsize, ysize, zsize
             ("xras", "f8", (3, 1)),  # x_r, x_a, x_s
             ("yras", "f8", (3, 1)),  # y_r, y_a, y_s
             ("zras", "f8", (3, 1)),  # z_r, z_a, z_s
-            ("cras", "f8", (3, 1)),  # c_r, c_a, c_s
+            ("cras", "f8", (3, )),  # c_r, c_a, c_s
             ("filename", "U1024"),
         ]
     )  # Not conformant (may be >1024 bytes)
     dtype = template_dtype
 
     def as_affine(self):
         """Return the internal affine of this regular grid."""
-        affine = np.eye(4)
         sa = self.structarr
-        A = np.hstack((sa["xras"], sa["yras"], sa["zras"])) * sa["voxelsize"]
-        b = sa["cras"] - A.dot(sa["volume"]) / 2
-        affine[:3, :3] = A
-        affine[:3, [3]] = b
-        return affine
+        R = np.hstack((sa["xras"], sa["yras"], sa["zras"])) * sa["voxelsize"]
+        offset = sa["cras"] - R @ sa["volume"] / 2
+        return from_matvec(R, offset)
```

Comment:
We should use a consistent notation here and in L74-80. I prefer A/b because this is a common representation of an augmented matrix (see [Affine transformation - Representation](https://en.wikipedia.org/wiki/Affine_transformation#Representation)).

In any event, I wouldn't use R, because that means rotation, and this matrix has scale and rotation. (Not sure if you can sneak shear into the MGH format in practice.)

```suggestion
        A = np.hstack((sa["xras"], sa["yras"], sa["zras"])) * sa["voxelsize"]
        b = sa["cras"] - A @ sa["volume"] / 2
        return from_matvec(A, b)
```